From d0c026907f53bbcf1eff1c547c8457a5c22d3b5d Mon Sep 17 00:00:00 2001
From: Xing Zheng <zhengxing@rock-chips.com>
Date: Fri, 20 Apr 2018 11:33:32 +0800
Subject: [PATCH] arm64: dts: rockchip: move acodec sound node to
 rk3308-evb-v10.dtsi

The acodec part is shared with the RK3308 EVB, so that users
can select an external sound card device as required.

Change-Id: I42c1de879ba14a693c161a401ecf5ba6a8d39cd0
Signed-off-by: Xing Zheng <zhengxing@rock-chips.com>
---
 .../boot/dts/rockchip/rk3308-evb-amic-v10.dts | 28 -------------------
 .../boot/dts/rockchip/rk3308-evb-v10.dtsi     | 28 +++++++++++++++++++
 2 files changed, 28 insertions(+), 28 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3308-evb-amic-v10.dts b/arch/arm64/boot/dts/rockchip/rk3308-evb-amic-v10.dts
index 146e130bd87f..8b460e42e562 100644
--- a/arch/arm64/boot/dts/rockchip/rk3308-evb-amic-v10.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3308-evb-amic-v10.dts
@@ -11,32 +11,4 @@
 / {
 	model = "Rockchip RK3308 evb analog mic board";
 	compatible = "rockchip,rk3308-evb-amic-v10", "rockchip,rk3308";
-
-	sound {
-		compatible = "simple-audio-card";
-		simple-audio-card,format = "i2s";
-		simple-audio-card,name = "rockchip,rk3308-acodec";
-		simple-audio-card,mclk-fs = <256>;
-		simple-audio-card,cpu {
-			sound-dai = <&i2s_8ch_2>;
-		};
-		simple-audio-card,codec {
-			sound-dai = <&acodec>;
-		};
-	};
-};
-
-&acodec {
-	status = "okay";
-
-	#sound-dai-cells = <0>;
-
-	hp-ctl-gpios = <&gpio0 RK_PA1 GPIO_ACTIVE_HIGH>;
-	spk-ctl-gpios = <&gpio0 RK_PA5 GPIO_ACTIVE_HIGH>;
-};
-
-&i2s_8ch_2 {
-	status = "okay";
-
-	#sound-dai-cells = <0>;
 };
diff --git a/arch/arm64/boot/dts/rockchip/rk3308-evb-v10.dtsi b/arch/arm64/boot/dts/rockchip/rk3308-evb-v10.dtsi
index c731d4c0ca69..b7af7fa81785 100644
--- a/arch/arm64/boot/dts/rockchip/rk3308-evb-v10.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3308-evb-v10.dtsi
@@ -102,6 +102,19 @@
 		reset-gpios = <&gpio0 RK_PA2 GPIO_ACTIVE_LOW>;
 	};
 
+	sound {
+		compatible = "simple-audio-card";
+		simple-audio-card,format = "i2s";
+		simple-audio-card,name = "rockchip,rk3308-acodec";
+		simple-audio-card,mclk-fs = <256>;
+		simple-audio-card,cpu {
+			sound-dai = <&i2s_8ch_2>;
+		};
+		simple-audio-card,codec {
+			sound-dai = <&acodec>;
+		};
+	};
+
 	spdif_tx_sound: spdif-tx-sound {
 		status = "disabled";
 		compatible = "simple-audio-card";
@@ -227,6 +240,15 @@
 	};
 };
 
+&acodec {
+	status = "okay";
+
+	#sound-dai-cells = <0>;
+
+	hp-ctl-gpios = <&gpio0 RK_PA1 GPIO_ACTIVE_HIGH>;
+	spk-ctl-gpios = <&gpio0 RK_PA5 GPIO_ACTIVE_HIGH>;
+};
+
 &cpu0 {
 	cpu-supply = <&vdd_core>;
 };
@@ -608,6 +630,12 @@
 	};
 };
 
+&i2s_8ch_2 {
+	status = "okay";
+
+	#sound-dai-cells = <0>;
+};
+
 &nandc {
 	status = "okay";
 };
-- 
2.35.3

